File: 682-棒球比赛.py
from typing import List


class Solution:
    def calPoints(self, ops: List[str]) -> int:
        res = []
        for ch in ops:
            if ch == '+':
                res.append(res[-1]+res[-2])
            elif ch == 'D':
                res.append(2*res[-1])
            elif ch == 'C':
                res.pop()
            else:
                res.append(int(ch))
        return sum(res)


ops = ["5", "-2", "4", "C", "D", "9", "+", "+"]
sol = Solution().calPoints(ops)
print(sol)

File: 682-棒球比赛.cpp
#include <iostream>
using namespace std;
#include <vector>
#include <string>
#include <numeric>
class Solution
{
public:
    int calPoints(vector<string> &ops)
    {
        vector<int> res;
        for (auto &ch : ops)
        {
            int n = res.size();
            switch (ch[0])
            {
            case '+':
                res.push_back(res[n - 1] + res[n - 2]);
                break;
            case 'D':
                res.push_back(2 * res[n - 1]);
                break;
            case 'C':
                res.pop_back();
                break;
            default:
                res.push_back(stoi(ch));
                break;
            }
        }
        return accumulate(res.begin(), res.end(), 0);
    }
};